doc/stomp_planner/stomp_planner_tutorial.rst

```diff
@@ -0,0 +1,221 @@
+
+#. Adjust the line ``<rosparam command="load" file="$(find panda_moveit_config)/config/stomp_planning.yaml" />`` to ``<rosparam command="load" file="$(find <robot_moveit_config>)/config/stomp_planning.yaml" />`` replacing ``<robot_moveit_config>`` with the name of your MoveIt! configuration package.
+#. Download `stomp_planning.yaml <https://github.com/ros-planning/panda_moveit_config/blob/master/config/stomp_planning.yaml>`_ file into the config directory of your MoveIt! config package. In our case, we will save this file in the ``panda_moveit_config/config`` directory. Create the "*stomp_planning.yaml*" configuration file. This file contains the parameters required by STOMP.  The parameters are specific to each ''planning group'' defined in   the SRDF file.  So if there are three planning groups, then the configuration file defines a specific set of parameters for each  planning group. In our case there is only one planning group, the "panda_arm": ::
+
+    stomp/manipulator_rail:
+      group_name: panda_arm
+      optimization:
+        num_timesteps: 60
+        num_iterations: 40
+        num_iterations_after_valid: 0    
+        num_rollouts: 30
+        max_rollouts: 30 
+        initialization_method: 1 #[1 : LINEAR_INTERPOLATION, 2 : CUBIC_POLYNOMIAL, 3 : MININUM_CONTROL_COST
+        control_cost_weight: 0.0
+      task:
+        noise_generator:
+          - class: stomp_moveit/NormalDistributionSampling
+            stddev: [0.05, 0.8, 1.0, 0.8, 0.4, 0.4, 0.4]
+        cost_functions:
+          - class: stomp_moveit/CollisionCheck
+            collision_penalty: 1.0
+            cost_weight: 1.0
+            kernel_window_percentage: 0.2
+            longest_valid_joint_move: 0.05 
+        noisy_filters:
+          - class: stomp_moveit/JointLimits
+            lock_start: True
+            lock_goal: True
+          - class: stomp_moveit/MultiTrajectoryVisualization
+            line_width: 0.02
+            rgb: [255, 255, 0]
+            marker_array_topic: stomp_trajectories
+            marker_namespace: noisy
+        update_filters:
+          - class: stomp_moveit/PolynomialSmoother
+            poly_order: 6
+          - class: stomp_moveit/TrajectoryVisualization
+            line_width: 0.05
+            rgb: [0, 191, 255]
+            error_rgb: [255, 0, 0]
+            publish_intermediate: True
+            marker_topic: stomp_trajectory
+            marker_namespace: optimized      
+    
+   **>** *Save this file in the* **config** *directory of the moveit_config package*. Also make sure that the dimensionality of the `stddev` array parameter is the same as the number of joints present in the planning group name of your robot.
+
+#. Modify the **move_group.launch** file. Open the **move_group.launch** in the launch directory and change the ```pipeline``` parameter value to ```stomp``` as shown below: ::
+
+        .
```

Comment:
why so many line breaks?

Reply:
I removed any extra line breaks. I have one line break just before the dot (.) , dot (.) , dot (.) , the dots are just to give a feeling to the user that this line is present somewhere in the middle in the [move_group.launch file](https://github.com/ros-planning/panda_moveit_config/blob/master/launch/move_group.launch#L27) . I can remove these dots if you say so.

Reply:
removed the extra dots